rust/src/dataset.rs

```diff
@@ -147,6 +140,13 @@ impl Dataset {
                         // offsets
                         let cur_src_pos = cur_box.min() - dst_pos;
                         let cur_dst_pos = cur_box.min() - cur_file_box.min();
+                        if self.header.block_type != BlockType::Raw {  
```

Comment:
Please use `self.header.block_type == BlockType::LZ4 || self.header.block_type == BlockType::LZ4HC`` instead. This is correct at the moment, but might cause trouble as soon as we add a new non-LZ4 block type.